tests/unit/test_store/fake_storage.py

```diff
@@ -1,107 +1,244 @@
+               terminate_after: Optional[Any] = ..., timeout: Optional[Any] = ..., track_scores: Optional[Any] = ...,
+               track_total_hits: Optional[Any] = ..., typed_keys: Optional[Any] = ..., version: Optional[Any] = ...,
+               pretty: Optional[bool] = ..., human: Optional[bool] = ..., error_trace: Optional[bool] = ...,
+               format: Optional[str] = ..., filter_path: Optional[Union[str, Collection[str]]] = ...,
+               request_timeout: Optional[Union[int, float]] = ..., ignore: Optional[Union[int, Collection[int]]] = ...,
+               opaque_id: Optional[str] = ..., params: Optional[MutableMapping[str, Any]] = ...,
+               headers: Optional[MutableMapping[str, str]] = ...) -> Any:
+        return "FakeSearchResult"
+
+    def get(self, index: Any, id: Any, *, doc_type: Optional[Any] = ..., _source: Optional[Any] = ...,
+            _source_excludes: Optional[Any] = ..., _source_includes: Optional[Any] = ...,
+            preference: Optional[Any] = ..., realtime: Optional[Any] = ..., refresh: Optional[Any] = ...,
+            routing: Optional[Any] = ..., stored_fields: Optional[Any] = ..., version: Optional[Any] = ...,
+            version_type: Optional[Any] = ..., pretty: Optional[bool] = ..., human: Optional[bool] = ...,
+            error_trace: Optional[bool] = ..., format: Optional[str] = ...,
+            filter_path: Optional[Union[str, Collection[str]]] = ...,
+            request_timeout: Optional[Union[int, float]] = ..., ignore: Optional[Union[int, Collection[int]]] = ...,
+            opaque_id: Optional[str] = ..., params: Optional[MutableMapping[str, Any]] = ...,
+            headers: Optional[MutableMapping[str, str]] = ...) -> Any:
+        return "FakeGetResult"
+
+    def index(self, index: Any, *, body: Any, doc_type: Optional[Any] = ..., id: Optional[Any] = ...,
+              if_primary_term: Optional[Any] = ..., if_seq_no: Optional[Any] = ..., op_type: Optional[Any] = ...,
+              pipeline: Optional[Any] = ..., refresh: Optional[Any] = ..., require_alias: Optional[Any] = ...,
+              routing: Optional[Any] = ..., timeout: Optional[Any] = ..., version: Optional[Any] = ...,
+              version_type: Optional[Any] = ..., wait_for_active_shards: Optional[Any] = ...,
+              pretty: Optional[bool] = ..., human: Optional[bool] = ..., error_trace: Optional[bool] = ...,
+              format: Optional[str] = ..., filter_path: Optional[Union[str, Collection[str]]] = ...,
+              request_timeout: Optional[Union[int, float]] = ..., ignore: Optional[Union[int, Collection[int]]] = ...,
+              opaque_id: Optional[str] = ..., params: Optional[MutableMapping[str, Any]] = ...,
+              headers: Optional[MutableMapping[str, str]] = ...) -> Any:
+        return "FakeIndexResult"
+
+
+class FakeEsPandasClient(es_pandas):
+
+    def __init__(self):
+        self.store = FakeIndexStore()

```

Comment:
we are getting funky here! 
a chain of fakes using fakes! A fake chain! Although, NO, the chain is true, and the fakes are true too. :)